arpa-exporter/src/worker.py

```diff
@@ -10,77 +13,129 @@
-                entry_logger.info("added file to archive")
-            else:
-                # TODO should we log that it already exists?
-                entry_logger.info("file already exists in archive")
+                entry_logger.info("Added file to the archive.")
 
     logger.info("updated zip archive", files_added=files_added)
 
 
-def process_sqs_message_request(s3, message_data, local_file):
+def load_source_paths_from_csv(s3, bucket, file_key: str):
+    response = s3.get_object(
+            Bucket=bucket,
+            Key=file_key,
+        )
+    bytes_stream = response["Body"]
+    csv_file_stream = io.TextIOWrapper(bytes_stream, encoding="utf-8")
+    reader = csv.DictReader(
+        csv_file_stream, delimiter=","
+    )
+
+    for row in reader:
+        yield UploadInfo(**row)
+
+
+def build_and_send_email(email_client, user_email: str, download_link: str):
+    logger = get_logger()
+    email_html, email_text, subject = generate_email(logger, download_link)
+    send_email(
+        email_client,
+        dest_email=user_email,
+        email_html=email_html or "",
+        email_text=email_text or "",
+        subject=subject or "",
+        logger=logger,
+    )
+
+def process_sqs_message_request(s3, ses, message_data: MessageSchema, local_file):
     # Get the S3 object if it already exists (if 404, assume it doesn't & create from scratch)
     # Note:
     #   Fargate ephemeral storage is free up to 20GB, so we should download and/or build
     #   the S3 file using tempfile storage. We have a good pattern for doing that using context
     #   managers in CPF Reporter.
-    s3_bucket = message_data["s3"]["bucket"]
-    s3_key = message_data["s3"]["key"]
+    s3_bucket = message_data.s3.bucket
+    s3_key = message_data.s3.zip_key
 
+    # Step 1 - Download the existing zip archive from S3 if exists
     try:
         s3.download_fileobj(s3_bucket, s3_key, local_file)
     except botocore.exceptions.ClientError as e:
         if e.response["Error"]["Code"] != "404":
             get_logger().exception("error downloading S3 object")
             raise
 
-    # TODO: use message_data to determine source files that should be in the archive
+    # Step 2 - Add or update contents of the zipfile with the new metadata
     try:
-        build_zip(local_file, source_paths=message_data["sources"])
+        build_zip(s3, local_file, message_data.s3.bucket, message_data.s3.metadata_key)
     except:
         get_logger().exception("error building zip archive")
         raise
 
+    # Step 3 - Upload the zip archive back to S3
     try:
         local_file.seek(0)
         s3.upload_fileobj(local_file, s3_bucket, s3_key)
     except:
         get_logger().exception("error uploading zip archive to s3")
         raise
 
+    # Step 4 - Send Email to user with the download link
+    try:
+        build_and_send_email(ses, message_data.user_email, download_link=f"/api/uploads/{message_data.organization_id}/{s3_key.split('/')[-1]}")
```

Comment:
change this to use `message_data.s3.metadata_key`
```suggestion
        build_and_send_email(ses, message_data.user_email, download_link=f"/api/uploads/{message_data.organization_id}/{message_data.s3.metadata_key.split('/')[-1]}")
```